cpp/src/arrow/c/helpers.h

```diff
@@ -18,10 +18,20 @@
 #pragma once
 
 #include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 
 #include "arrow/c/abi.h"
 
+#define CHECK_OR_DIE(condition, msg)                          \
+  do {                                                        \
+    if (!(condition)) {                                       \
+      fprintf(stderr, "%s:%d:: %s", __FILE__, __LINE__, msg); \
+      std::abort();                                           \
```

Comment:
This needs to be C code, not C++.

Reply:
good point, my mistake there at using `std::` it should be just `abort`.